src/test/java/org/folio/rest/api/DereferencedItemStorageTest.java

```diff
@@ -0,0 +1,245 @@
+    response = getCompleted.get(5, SECONDS);
+    assertThat(response.getStatusCode(), is(201));
+  }
+
+  @AfterClass
+  public static void cleanUpDatabase() {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+  }
+
+  @Test
+  public void CanGetRecordByCQLSearch() {
+    String queryString = "barcode=036000291452";
+    
+    DereferencedItems items = findByCql(queryString);
+
+    assertThat(items.getTotalRecords(), is(1));
+
+    DereferencedItem item = items.getDereferencedItems().get(0);
+
+    assertThat(item.getBarcode(), is("036000291452"));
+    assertThat(item.getId(), is(smallAngryPlanetId.toString()));
+    assertThat(item.getInstanceRecord().getTitle(), is("Long Way to a Small Angry Planet"));
+    assertThat(item.getPermanentLoanType().getName(), is("Can Circulate"));
+    assertThat(item.getMaterialType().getName(), is("journal"));
+    assertThat(item.getHoldingsRecord().getInstanceId(), is(item.getInstanceRecord().getId()));
+  }
+
+  public void ResturnsAllRecordsWhenNoCQLQuery() {
+    DereferencedItems items = getAll();
+
+    assertThat(items.getTotalRecords(), is(3));
+  }
+
+  public void Returns404WhenNoItemsFound() {
+    String queryString = "barcode=647671342075";
+    
+    Response response = attemptFindByCql(queryString);
+
+    assertThat(response.getStatusCode(), is(404));
+  }
+
+  public void Returns400WhenCqlSearchInvalid() {
+    String queryString = "barcode&647671342075";
+    
+    Response response = attemptFindByCql(queryString);
+
+    assertThat(response.getStatusCode(), is(400));
+  }
+
+  public void CanGetRecordById() {
+    DereferencedItem item = findById(smallAngryPlanetId.toString());
+
+    assertThat(item.getBarcode(), is("036000291452"));
+    assertThat(item.getId(), is(smallAngryPlanetId.toString()));
+    assertThat(item.getInstanceRecord().getTitle(), is("Long Way to a Small Angry Planet"));
+    assertThat(item.getPermanentLoanType().getName(), is("Can Circulate"));
+    assertThat(item.getMaterialType().getName(), is("journal"));
+    assertThat(item.getHoldingsRecord().getInstanceId(), is(item.getInstanceRecord().getId()));
+  }
+
+  public void Returns404WhenNoItemFoundForId() {
+    String Id = UUID.randomUUID().toString();
+    
+    Response response = attemptFindById(Id);
+
+    assertThat(response.getStatusCode(), is(404));
+  }
+
+  public void Returns400WhenInvalidUUID() {
+    String Id = "w325b3dc4";
+    
+    Response response = attemptFindById(Id);
+
+    assertThat(response.getStatusCode(), is(400));
+  }
+
+
+  private static JsonObject createItemRequest(
+      UUID id,
+      UUID holdingsRecordId,
+      String barcode) {
+
+    return createItemRequest(id, holdingsRecordId, barcode, journalMaterialTypeID);
+  }
+
+  private static JsonObject createItemRequest(
+    UUID id,
+    UUID holdingsRecordId,
+    String barcode,
+    String materialType) {
```

Comment:
Why is every parameter on a separate line?

Reply:
Thank you, that formatting style was likely something I did a long time ago and was a mistake.